src/test/java/org/cryptomator/common/SubstitutingPropertiesTest.java

```diff
@@ -0,0 +1,151 @@
+package org.cryptomator.common;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.Mockito;
+
+import java.util.Map;
+import java.util.Properties;
+
+public class SubstitutingPropertiesTest {
+
+	SubstitutingProperties inTest;
+
+	@Nested
+	public class Processing {
+
+		@ParameterizedTest
+		@DisplayName("Test template replacement")
+		@CsvSource(textBlock = """
+				unknown.@{testToken}.test, unknown.@{testToken}.test
+				@{only*words*digits*under_score},@{only*words*digits*under_score}
+				C:\\Users\\@{appdir}\\dir, C:\\Users\\foobar\\dir
+				@{@{appdir}},@{foobar}
+				Replacing several @{appdir} with @{appdir}., Replacing several foobar with foobar.""")
+		public void test(String propertyValue, String expected) {
+			SubstitutingProperties inTest = new SubstitutingProperties(Mockito.mock(Properties.class), Map.of("APPDIR", "foobar"));
+			var result = inTest.process(propertyValue);
+			Assertions.assertEquals(result, expected);
```

Comment:
should be:

```suggestion
			Assertions.assertEquals(expected, result);
```

(same in various other places)

Reply:
Fixed in  https://github.com/cryptomator/cryptomator/pull/2957/commits/29d63a0a83b7399ba65bd07ff57bf28e5a770ffd